src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1210,6 +1209,19 @@ impl LinkCollector<'_, '_> {
                         }
                     }
                 }
+
+                let is_pointer =
+                    prim == PrimitiveType::RawPointer || prim == PrimitiveType::Reference;
+                if is_pointer && !self.cx.tcx.features().intra_doc_pointers {
+                    rustc_session::parse::feature_err(
+                        &self.cx.tcx.sess.parse_sess,
+                        sym::intra_doc_pointers,
+                        span_of_attrs(&item.attrs).unwrap_or(item.source.span()),
+                        "rustdoc does not allow disambiguating `*const` and `*mut`, and pointers/references are unstable until it does",
```

Comment:
What does "rustdoc does not allow disambiguating `*const` and `*mut`" mean? My understanding is this error should be something like:

```suggestion
                        "raw pointer and reference intra-doc link targets are unstable",
```

Reply:
Also, are we sure that we want to feature-gate the `*const` and `*mut` targets too?

Reply:
> Maybe say "disambiguating between" to make it clearer?

Good idea, done.

> Also, are we sure that we want to feature-gate the *const and *mut targets too?

You mean without any associated items? Hmm, it seems reasonable to allow those on their own, since there's only one page for both. I'll change that.

Reply:
> > Also, are we sure that we want to feature-gate the *const and *mut targets too?
> 
> You mean without any associated items? Hmm, it seems reasonable to allow those on their own, since there's only one page for both. I'll change that.

Actually when I wrote that comment I forgot that the reason you feature-gated `*const` and `*mut` was primarily because we can't disambiguate between them 😅 

It actually may still be good to feature-gate `*const` and `*mut` (but keep `pointer` as non-feature-gated) after all, for one because we may want to change the syntax and thus shouldn't preemptively stabilize. Sorry to send you in circles!

Reply:
I don't think we'll want to change that syntax; what would we replace it with?

Reply:
I guess I was thinking we could do `*const T` or something, but that probably wouldn't be great. It's probably fine to stabilize, but I just want to make sure we're careful since once we stabilize we can't remove it :)

Reply:
I think the current syntax is fine. The reason for feature gating associated items is because they're ambiguous and rustdoc can't warn about them, not because the syntax is strange.